src/test/java/com/alkemy/ong/bigtest/auth/RegisterAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,235 @@
+            .content(objectMapper.writeValueAsString(RegisterRequest.builder()
+                    .firstName("Pepe")
+                .lastName("Le Pew")
+                .password("pass1234")
+                .build()))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.statusCode", equalTo(400)))
+        .andExpect(jsonPath("$.message", equalTo("Invalid input data.")))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo",
+            hasItem("must not be null")))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  public void shouldReturnConflictWhenEmailIsAlreadyExist() throws Exception {
+
+    mockMvc.perform(post("/auth/register")
+            .content(objectMapper.writeValueAsString(RegisterRequest.builder()
+                .firstName("Pepe")
+                .lastName("Le Pew")
+                .email("freedy@krueger.com")
+                .password("pass1234")
+                .build()))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.statusCode", equalTo(409)))
+        .andExpect(jsonPath("$.message", equalTo("Email is already in use.")))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo",
+            hasItem("The server could not complete the user registration because the email address"
+                + " entered is already in use.")))
+        .andExpect(status().isConflict());
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenEmailHaveInvalidFormat() throws Exception {
+
+    mockMvc.perform(post("/auth/register")
+            .content(objectMapper.writeValueAsString(RegisterRequest.builder()
+                .firstName("Pepe")
+                .lastName("Le Pew")
+                .email("invalidFormat")
+                .password("pass1234")
+                .build()))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.statusCode", equalTo(400)))
+        .andExpect(jsonPath("$.message", equalTo("Invalid input data.")))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo",
+            hasItem("The email has invalid format.")))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenPasswordIsNull() throws Exception {
+
+    mockMvc.perform(post("/auth/register")
+            .content(objectMapper.writeValueAsString(RegisterRequest.builder()
+                .firstName("Pepe")
+                .lastName("Le Pew")
+                .email("pepeLePew@mail.com")
+                .build()))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.statusCode", equalTo(400)))
+        .andExpect(jsonPath("$.message", equalTo("Invalid input data.")))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo",
+            hasItem("must not be null")))
```

Comment:
Hay un bug en el DTO porque "must not be null" no es un mensaje de error representativo, debería indicar el nombre del campo. Ejemplo: "firstName must not be null."